src/charm.py

```diff
@@ -903,6 +898,11 @@ def pass_pre_set_password_checks(self, event: ActionEvent) -> bool:
             event.fail("Cannot set passwords while an upgrade is in progress.")
             return False
 
+        if self.upgrade_in_progress:
+            logger.debug("Do not set the password while a backup/restore is in progress.")
+            event.fail("Cannot set passwords while an upgrade is in progress.")
+            return False
```

Comment:
this block seems to be a duplicate of the previous check

Reply:
ah merge error, ty